src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1481,13 +1481,49 @@ private function registerAnnotationsConfiguration(array $config, ContainerBuilde
                 ->setMethodCalls([['registerLoader', ['class_exists']]]);
         }
 
-        if ('none' !== $config['cache']) {
+        if ('none' === $config['cache']) {
+            $container->removeDefinition('annotations.cached_reader');
+
+            return;
+        }
+
+        $cacheService = $config['cache'];
+        if (\in_array($config['cache'], ['php_array', 'file'])) {
+            $isPsr6Service = $container->hasDefinition('annotations.psr_cached_reader');
+        } else {
+            $isPsr6Service = false;
+            trigger_deprecation('symfony/framework-bundle', '5.3', 'Using a custom service for "framework.annotation.cache" is deprecated, only values "none", "php_array" and "file" are valid in version 6.0.');
+        }
+
+        if ($isPsr6Service) {
+            $container->removeDefinition('annotations.cached_reader');
+            $container->setDefinition('annotations.cached_reader', $container->getDefinition('annotations.psr_cached_reader'));
```

Comment:
shouldn't this set an alias rather than registering the same Definition object under 2 ids (which might lead to weird bugs as they are still the same definition) ?

Reply:
Great idea!